django/contrib/admin/models.py

```diff
@@ -44,6 +51,53 @@ def log_action(
             change_message=change_message,
         )
 
+    def log_actions(
+        self, user_id, queryset, action_flag, change_message="", *, single_object=False
+    ):
+        """
+        Method to bulk_create multiple LogEntry objects from a given queryset.
+        """
+        # If log_action is overridden in a subclass, call it to maintain
+        # backward compatibility
+        if type(self).log_action != LogEntryManager.log_action:
+            return [
+                self.log_action(
+                    user_id=user_id,
+                    content_type_id=ContentType.objects.get_for_model(
+                        obj, for_concrete_model=False
+                    ).id,
+                    object_id=str(obj.pk),
+                    object_repr=repr(obj),
```

Comment:
> I have tried both, many past tests get failed in case of `str(obj)`. The `repr(obj)` is comparatively safe.

How it is possible? Previously it used `str()` so using it here should not cause any failures.

Reply:
Sorry those were my earlier changes failing the test. Reverted them.